Detectors/TRD/base/src/TrackletTransformer.cxx

```diff
@@ -0,0 +1,127 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "DetectorsBase/GeometryManager.h"
+#include "TRDBase/TrackletTransformer.h"
+#include "TMath.h"
+
+using namespace o2::trd;
+using namespace o2::trd::constants;
+
+TrackletTransformer::TrackletTransformer()
+{
+  o2::base::GeometryManager::loadGeometry();
+  mGeo = Geometry::instance();
+  mGeo->createPadPlaneArray();
+  mGeo->createClusterMatrixArray();
+}
+
+void TrackletTransformer::loadPadPlane(int hcid)
+{
+  int detector = hcid / 2;
+  int stack = mGeo->getStack(detector);
+  int layer = mGeo->getLayer(detector);
+  mPadPlane = mGeo->getPadPlane(layer, stack);
+}
+
+float TrackletTransformer::calculateX()
+{
+  // report x at 5mm below the cathode plan to minimize error propogration from tracklet fit and driftV calibration
+  float driftHeight = mGeo->cdrHght();
+  float x = driftHeight - 0.5;
+
+  return x;
+}
+
+float TrackletTransformer::calculateY(int hcid, int column, int position)
+{
+  double padWidth = mPadPlane->getWidthIPad();
+  int side = hcid % 2;
+
+  // slightly modified TDP eq 16.1 (appended -1 to the end to account for MCM shared pads)
+  double pad = float(position - (1 << (NBITSTRKLPOS - 1))) * GRANULARITYTRKLPOS + NCOLMCM * (4 * side + column) + 10.5 - 1;
+  float y = padWidth * (pad - 72);
+
+  return y;
+}
+
+float TrackletTransformer::calculateZ(int padrow)
+{
+  double rowPos = mPadPlane->getRowPos(padrow);
+  double rowWidth = mPadPlane->getRowSize(padrow);
+  double middleRowPos = mPadPlane->getRowPos(mPadPlane->getNrows() / 2);
+
+  float z = rowPos - rowWidth / 2. - middleRowPos;
+
+  return z;
+}
+
+float TrackletTransformer::calculateDy(int slope)
```

Comment:
calculateDy does not seem to include ExB at the moment. Is that intentional?

Reply:
Ok, I found the Lorentz angle below. I am however not sure why this is not done in one step? v_drift and ExB seem very closely connected, and I would expect that they are best treated at the same time.

Reply:
Good point. I'll merge `TrackletTransformer::calculateDy()` and `TrackletTransformer::calibrateDy()` into one function